.github/workflows/test.yml

```diff
@@ -12,51 +12,22 @@ on:
     types: [ released ]
 
 jobs:
-  flake8:
-    name: flake8
+  linting:
     runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.10'
-      - run: python -m pip install flake8
-      - name: flake8
-        uses: liskin/gh-problem-matcher-wrap@v1
-        with:
-          linters: flake8
-          run: flake8
-
-  isort:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.10'
-      - run: python -m pip install isort
-      - name: isort
-        uses: liskin/gh-problem-matcher-wrap@v1
-        with:
-          linters: isort
-          run: isort --check --profile black src/
+    strategy:
+      matrix:
+        python-version: ["3.11"]
 
-  black:
-    runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Set up Python
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
-          python-version: '3.10'
-      - run: python -m pip install black
-      - name: black
-        run: black --check src/
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: pip install pre-commit
+      - name: Run pre-commit
+        run: pre-commit run --all-files
```

Comment:
Isn't this changing the source files?

Pre-commit should also be used optionally, locally for committing purposes in my opinion. The CI then validates the checks independently.

Reply:
This is indeed modifying source files is something is wrong, but will abort the execution.

I think `precommit` has multiple checks that were not run separately (for example removing trailing spaces). I think `precommit` should not be optional and mandatory for everyone collaborating to this project so it's more convenient to just run in on the CI instead of configuring multiple checks separately